src/components/shared/Camera.tsx

```diff
@@ -0,0 +1,183 @@
+ * in the lowest quality video available with a 4:3 aspect ratio with a 10 second cap.
+ */
+
+import * as React from "react";
+import { StyleSheet, View, Text, TouchableOpacity, Button } from "react-native";
+import { Camera as ExpoCamera, Permissions, CameraObject } from "expo";
+
+type CameraProps = {
+  onVideoRecorded: (uri: string) => any;
+};
+
+interface CameraState {
+  hasCameraPermission?: boolean;
+  hasAudioRecordPermission?: boolean;
+  type: string;
+  isVideoRecording: boolean;
+}
+
+class Camera extends React.Component<CameraProps, CameraState> {
+  private cameraRef?: React.RefObject<CameraObject & ExpoCamera>;
+
+  state = {
+    hasCameraPermission: undefined,
+    hasAudioRecordPermission: undefined,
+    type: "front",
+    isVideoRecording: false
+  };
+
+  constructor(props: CameraProps) {
+    super(props);
+    this.cameraRef = React.createRef();
+  }
+
+  async componentWillMount() {
+    await this.requestPermissions();
+  }
+
+  async requestPermissions() {
+    const results = await Promise.all([
+      Permissions.askAsync(Permissions.CAMERA),
+      Permissions.askAsync(Permissions.AUDIO_RECORDING)
+    ]);
+    this.setState({
+      hasCameraPermission: results[0].status === "granted",
+      hasAudioRecordPermission: results[1].status === "granted"
+    });
+  }
+
+  startRecordVideo = (camera: CameraObject) => {
+    camera
+      .recordAsync({
+        // quality: Camera.Constants.VideoQuality['720p'],
+        maxDuration: 10 // seconds
+      })
+      .then(({ uri }) => {
+        this.setState({
+          isVideoRecording: false
+        });
+        this.props.onVideoRecorded(uri);
+      })
+      .catch(reason => {
+        // TODO: Handle error, toast a message?
+      });
+
+    this.setState({
+      isVideoRecording: true
+    });
+  };
+
+  render() {
+    if (
+      !this.state.hasCameraPermission ||
+      !this.state.hasAudioRecordPermission
+    ) {
+      return (
+        <View>
+          <Text>
+            In order to verify your identity, you must record a video of
+            yourself. Please approve the permissions to continue.
+          </Text>
+          <Button
+            title="Approve Permissions"
+            onPress={() => {
+              this.requestPermissions();
+            }}
+          />
+        </View>
+      );
+    } else {
+      return (
+        <View style={styles.container}>
+          <ExpoCamera
+            style={styles.camera}
+            type={this.state.type}
+            ref={this.cameraRef}
+          >
+            <View style={styles.cameraButtons}>
+              {this.renderFlipButton()}
+              {this.renderRecordButton()}
+            </View>
+          </ExpoCamera>
+        </View>
+      );
+    }
+  }
+
+  renderFlipButton = () => {
+    return (
+      <TouchableOpacity
+        style={styles.flipButton}
+        onPress={() => {
+          this.setState({
+            type: this.state.type === "back" ? "front" : "back"
```

Comment:
Maybe leave a quick TODO for posterity here that once the expo types include the constants we ought to use them here.